### PR TITLE
Issues

### PR DESCRIPTION
**Pull request by UsmanAhmad169**
This branch adds Issue and Comment models (along-with migrations). Also "aasm" gem is installed and integrated in the Issue model (after_commit_everywhere gem is also installed).

### CODE REVIEW

File: app/models/issue.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
```
Comment (mujeeb-arshad) [inline app/models/issue.rb]: remove empty line

File: config/database.yml [hunk 2]
```diff
@@ -1,54 +0,0 @@
-# MySQL. Versions 5.5.8 and up are supported.
-#
-# Install the MySQL driver
-#   gem install mysql2
```
Comment (mujeeb-arshad) [inline config/database.yml:4]: why this file is deleted?

File: db/migrate/20200907124126_create_comments.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class CreateComments < ActiveRecord::Migration[6.0]
+  def change
+    create_table :comments do |t|
+      t.references :commentable, polymorphic: true
+
```
Comment (mujeeb-arshad) [inline db/migrate/20200907124126_create_comments.rb]: remove extra lines

File: db/migrate/20200907140351_rename_type_column_in_issues_table.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class RenameTypeColumnInIssuesTable < ActiveRecord::Migration[6.0]
+  def change
+    rename_column :issues, :type, :issue_type
```
Comment (mujeeb-arshad) [inline db/migrate/20200907140351_rename_type_column_in_issues_table.rb]: rename the column in the previous migration by deleting from schema_migration / rollback. Only add migration of rename of change column if the db already merged. Discuss with me or your mentor.

File: db/schema.rb [hunk 5]
```diff
@@ -0,0 +1,34 @@
+# This file is auto-generated from the current state of the database. Instead
```
Comment (mujeeb-arshad) [inline db/schema.rb]: add this to git ignore as well

File: test/fixtures/comments.yml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
+
```
Comment (mujeeb-arshad) [inline test/fixtures/comments.yml]: delete the test folder

**Comment by mujeeb-arshad** [comment]
cycle 1

File: test/fixtures/comments.yml (hunk 6 above)
Comment (UsmanAhmad169) [reply test/fixtures/comments.yml]: Done

File: Gemfile [hunk 7]
```diff
@@ -22,6 +22,10 @@ gem 'jbuilder', '~> 2.7'
 # Use Active Model has_secure_password
 # gem 'bcrypt', '~> 3.1.7'
 
+# Gems for aasm support
+gem 'aasm'
```
Comment (muhammad-basil) [inline Gemfile]: add gem version

File: config/database.yml (hunk 2 above)
Comment (UsmanAhmad169) [reply config/database.yml:4]: Added again in newer commits

File: db/schema.rb (hunk 5 above)
Comment (UsmanAhmad169) [reply db/schema.rb]: Added to gitignore

File: Gemfile (hunk 7 above)
Comment (UsmanAhmad169) [reply Gemfile]: Done

File: db/migrate/20200907124126_create_comments.rb (hunk 3 above)
Comment (UsmanAhmad169) [reply db/migrate/20200907124126_create_comments.rb]: Done

File: app/models/issue.rb (hunk 1 above)
Comment (UsmanAhmad169) [reply app/models/issue.rb]: Done

File: db/migrate/20200907140351_rename_type_column_in_issues_table.rb (hunk 4 above)
Comment (UsmanAhmad169) [reply db/migrate/20200907140351_rename_type_column_in_issues_table.rb]: Doing it now

Comment (UsmanAhmad169) [reply db/migrate/20200907140351_rename_type_column_in_issues_table.rb]: Done

File: db/migrate/20200907122828_create_issues.rb [hunk 8]
```diff
@@ -0,0 +1,14 @@
+class CreateIssues < ActiveRecord::Migration[6.0]
+  def change
+    create_table :issues do |t|
+      t.string :title
+      t.text :description
+
+      t.integer :issue_type # enum: bug or issue
```
Comment (muhammad-basil) [inline db/migrate/20200907122828_create_issues.rb]: add null: false to all the columns you have added validations for

File: db/migrate/20200907122828_create_issues.rb [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+      t.integer :priority # enum: low or high
```
Comment (muhammad-basil) [inline db/migrate/20200907122828_create_issues.rb]: Add company's association to all the tables + add association of project with issues once you have latest code

File: db/migrate/20200907122828_create_issues.rb (hunk 8 above)
Comment (UsmanAhmad169) [reply db/migrate/20200907122828_create_issues.rb]: Doing it now

File: db/migrate/20200907122828_create_issues.rb (hunk 9 above)
Comment (UsmanAhmad169) [reply db/migrate/20200907122828_create_issues.rb]: ok

File: db/migrate/20200907122828_create_issues.rb (hunk 8 above)
Comment (UsmanAhmad169) [reply db/migrate/20200907122828_create_issues.rb]: Done

File: db/migrate/20200907122828_create_issues.rb (hunk 9 above)
Comment (UsmanAhmad169) [reply db/migrate/20200907122828_create_issues.rb]: Done

File: app/models/project.rb [hunk 10]
```diff
@@ -7,4 +7,6 @@ class Project < ApplicationRecord
 
   has_many :project_users
   has_many :users, through: :project_users
-end
\ No newline at end of file
+
```
Comment (muhammad-basil) [inline app/models/project.rb:10]: remove extra line

File: app/models/user.rb [hunk 11]
```diff
@@ -11,4 +11,6 @@ class User < ApplicationRecord
   has_many :team_users
   has_many :teams, through: :team_users
```
Comment (muhammad-basil) [inline app/models/user.rb]: remove extra line

File: db/migrate/20200909072302_create_issues.rb [hunk 12]
```diff
@@ -0,0 +1,21 @@
+class CreateIssues < ActiveRecord::Migration[6.0]
+  def change
+    create_table :issues do |t|
+      t.string :title, null: false
+      t.text :description
+
+      t.integer :issue_type, null: false # enum: bug or issue
+      t.integer :priority, null: false # enum: low or high
+      t.integer :status, null: false # enum: open, in_progress, resolved or closed
+
```
Comment (muhammad-basil) [inline db/migrate/20200909072302_create_issues.rb:10]: remove extra lines

File: db/migrate/20200909072537_create_comments.rb [hunk 13]
```diff
@@ -0,0 +1,12 @@
+class CreateComments < ActiveRecord::Migration[6.0]
+  def change
+    create_table :comments do |t|
+      t.references :commentable, polymorphic: true, index: true
+      t.references :users, null: false, index: true
+      t.text :body, null: false
+
```
Comment (muhammad-basil) [inline db/migrate/20200909072537_create_comments.rb]: remove extra lines

File: app/models/project.rb (hunk 10 above)
Comment (UsmanAhmad169) [reply app/models/project.rb:10]: This line gets added by default as my editors setting. Rubocop objects for not having last blank line.

**Pull request merged**